state/core_access.go

```diff
@@ -699,3 +705,17 @@ func withFee(fee Int) user.TxOption {
 	gasFee := sdktypes.NewCoins(sdktypes.NewCoin(app.BondDenom, fee))
 	return user.SetFeeAmount(gasFee)
 }
+
+// THIS IS A TEMPORARY SOLUTION!!!
+// unsetTx helps to fix issue in TxResponse marshaling. Marshaling TxReponse
+// fails because `TxResponse.Tx` is not empty but does not contain respective codec
+// for encoding using the standard `json.MarshalJSON()`. It becomes empty when
+// https://github.com/celestiaorg/celestia-core/issues/1281 will be merged.
+// The `TxResponse.Tx` contains the transaction that is sent to the cosmos-sdk in the form
+// in which it is processed there, so the user should not be aware of it.
+func unsetTx(txResponse *TxResponse) *TxResponse {
+	if txResponse != nil && txResponse.Tx != nil {
+		txResponse.Tx = nil
```

Comment:
I would expect this to be resolvable differently. but this is a workable workaround if it's only short-lived.

Reply:
This is the simplest and the quickest way of resolving this issue from the node side. I also think it's not the node issue as we expect to get results that can be converted to both JSON and proto formats.

Reply:
If that is a contract between app and node (I might not be aware of that), then the app behaviour needs to change (cc @cmwaters )

Reply:
Can we break this out into a separate issue how to deal with this long term before merging? I also would appreciate some input from the app team there.

Reply:
After finishing https://github.com/celestiaorg/celestia-core/issues/1281 the Tx field won't be needed and it will be unset(or even removed) on the celestia-app side

Reply:
nice, thanks!

Reply:
One more question on this (before I would approve the PR): what is the _concrete_ type here that gets back from app (in txResponse.Tx)? I would be surprised if it did not come with a json marshaller implemented (it likely just lives in a different file which is not imported).

Reply:
> One more question on this (before I would approve the PR): what is the concrete type here that gets back from app (in txResponse.Tx)? I would be surprised if it did not come with a json marshaller implemented (it likely just lives in a different file which is not imported).

The `txResponse.Tx` type is [types.Any](https://github.com/celestiaorg/cosmos-sdk/blob/02de31436e7616f9215aa79404b00d17fe5c0e5a/codec/types/any.go#L11) that apparently does have default json unmarshaler as it's not set in the [c-tor](https://github.com/celestiaorg/cosmos-sdk/blob/02de31436e7616f9215aa79404b00d17fe5c0e5a/codec/types/any.go#L59). So, when it comes to the marshaling the `compat`(compatibility) field is empty, which cause an error. I have a theory that proto message should be wrapped somewhere by [ProtoJsonPacker](https://github.com/celestiaorg/cosmos-sdk/blob/02de31436e7616f9215aa79404b00d17fe5c0e5a/codec/types/compat.go#L186) but haven't found the confirmation yet.

Reply:
> The txResponse.Tx type is [types.Any](https://github.com/celestiaorg/cosmos-sdk/blob/02de31436e7616f9215aa79404b00d17fe5c0e5a/codec/types/any.go#L11) that apparently

Yeah, I thought the concrete Tx type that gets into the Any-typed field might have implemented the json marshaller directly somehow but it does not sound like it actually does.